lib/src/main/java/com/auth0/android/lock/utils/SignUpField.java

```diff
@@ -0,0 +1,52 @@
+package com.auth0.android.lock.utils;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+
+import static com.auth0.android.lock.utils.CustomField.Storage.PROFILE_ROOT;
+
+public abstract class SignUpField implements Parcelable {
+
+    private final String key;
+    private final int storage;
+
+    SignUpField(@NonNull String key, @CustomField.Storage int storage) {
+        if (key.isEmpty()) {
+            throw new IllegalArgumentException("The key cannot be empty!");
```

Comment:
```suggestion
            throw new IllegalArgumentException("The key cannot be empty.");
```

Reply:
I fear if I change this it would fail if a test is associated. Let me check this locally first.

Reply:
I was right! Resolved on a separate commit.